### ISSUE TITLE
Problem building rpm for OpenSUSE

### ISSUE THREAD

**Issue by dsteuer** [emacs-ess/ESS#731]
Hi! 

Packager for OpenSUSE here.
This release is somewhat more demanding than any release the last 10 years :-)

I have 18.10-1 and the Makefile for lisp/ provided by Martin.
(btw 18.10.1 would be much easier, at least for me!)

Nevertheless I still run into problems using my "good old spec file":

First: If I invoke the build service using my spec I get this:

```
[   13s] ./installation.texi:117: warning: node up `Check Installation' in menu `Installation' and in sectioning `General Information' differ
[   13s] readme.texi:8: node `General Information' lacks menu item for `Check Installation' despite being its Up target
[   13s] make[1]: [Makefile:117: html/readme.html] Error 1 (ignored)
[   13s] ##                   ^^^^^^^^^^^^^ (design bug: "FIXME" in ess.texi)
```

So it seems readme.html is not created and consequently

```
[   13s] cp -p html/ess.html html/readme.html html/news.html /home/abuild/rpmbuild/BUILDROOT/emacs-ess-18.10.1-1.x86_64/usr/share/doc/packages/emacs-ess
[   13s] cp: cannot stat 'html/readme.html': No such file or directory
```

the build stops.

Second: I wanted to debug locally and just invoke make by hand, but

```
steuer@gaia:~/OBS/devel:languages:R:released/emacs-ess/ess-18.10.1> make
cd lisp; make all
make[1]: Verzeichnis „/home/steuer/OBS/devel:languages:R:released/emacs-ess/ess-18.10.1/lisp“ wird betreten
emacs -batch -no-site-file -no-init-file --directory . -f batch-byte-compile ess.el
Eager macro-expansion failure: (error "‘tramp-file-name-structure’ didn’t match!")
Eager macro-expansion failure: (error "Variable binding depth exceeds max-specpdl-size")
>>Error occurred processing ess.el: error (("‘tramp-file-name-structure’ didn’t match!"))  
make[1]: *** [Makefile:99: ess.elc] Fehler 1
make[1]: Verzeichnis „/home/steuer/OBS/devel:languages:R:released/emacs-ess/ess-18.10.1/lisp“ wird verlassen
make: *** [Makefile:9: all] Fehler 2
```

And here i have no idea what happens. The same make works inside the build service, but make
always worked for older releases.

my emacs:

```
steuer@gaia:~> emacs --version
GNU Emacs 26.1
```

Any ideas?

Detlef

**Comment by eddelbuettel** [comment]
> So it seems readme.html is not created and consequently

Yes. This is no longer the default but someone thought having docs is uncool. I now do

```bash
(cd doc; $(MAKE) all; $(MAKE) info; $(MAKE) pdf; $(MAKE) html; cd -)
```

I converted the Debian package a little earlier (and with some help by a fellow Debianer and member of the Emacs packages team) to an elpa packages. That now seems to work fine (I had left a bad/old `ess-site.el` around which we by now addressed) but I do need to manually create an correct `ess-pkg.el` with the current version (so that the elpa / package mechanism can do its magic).  Currently:

```el
(define-package "ess" "18.10-1" "Emacs Speaks Statistics" nil
  :url "http://ess.r-project.org" :keywords nil)
```

You can see my packging over here.  Not pretty as it is a decade old and 'evolved' but functional:
https://salsa.debian.org/edd/ess/

**Comment by jabranham** [comment]
This is because of these lines in doc/Makefile:

```
install-other-docs: pdf html
	-$(INSTALLDIR) $(DOCDIR)
	$(INSTALL) ess.pdf readme.pdf $(DOCDIR)
	$(INSTALL) html/ess.html html/readme.html html/news.html $(DOCDIR)
	$(INSTALL) refcard/refcard.pdf $(DOCDIR)
```

where by default the pdf and html files aren't built unless you build them.

A workaround for the time being is to do like Dirk does and build the pdf and html files explicitly, or you can patch out the `install-other-docs` dependency in the `install` target in `doc/Makefile` depending on if you want to distribute html and pdf versions of the documentation or not.

**Comment by jabranham** [comment]
This should be fixed by 9648181562b19118b73fab3aec6d1e524140d2a5 where we're bundling html and pdf documentation with the .tgz files. Until the next tarball is released, you'll still need the workaround I mentioned above.
